src/metrics.h

```diff
@@ -0,0 +1,775 @@
+/**
+ * @file   metrics.h
+ * @date   04/2019 - now
+ * @author Diogo Valada
+ * @brief  OpenQl circuit fidelity estimator
+ */
+
+//TODO: ATM, this does not support operations after measurement (eg. preparing again and reusing a qubit)
+//The above will produce undefined behaviour
+
+#ifndef METRICS_H
+#define METRICS_H
+
+#include <cmath> 
+#include <vector>
+#include <map>
+#include <random>
+#include <chrono>
+#include "gate.h"
+#include "utils.h" 
+#include "options.h"
+#include "platform.h"
+//#include "mapper.h"
+
+
+
+#define CYCLE_TIME 20 //TODO: gate duration is hardcoded for now
+
+#include <iostream>
+#include <sstream>
+#define PRINTER(x) my_print(x, #x)
+template <typename T>
+
+void my_print(std::vector<T> const &input, const char *id_name)
+{
+	std::stringstream output;
+	output << id_name << "(" << input.size() << ")= ";
+	for (auto const& i: input) {
+		output << i << " ";
+	}
+	// output << "\n";
+	IOUT(output.str());
+}
+
+namespace ql
+{
+// namespace metrics
+// {
+
+
+// ql::circuit random_circuit_generator(double Nqubits, size_t Ncycles, double idle_fraction, double two_qb_gate_fraction ){
```

Comment:
The scheduler diff is big because the pre179 code was inadvertently taken out
No harm because post179 is compatible and has proven itself
I propose to leave it like this

Reply:
cleaned some comments, leaving others as it seems work in progress